Parse.hs

```diff
@@ -11,15 +11,18 @@ import Types
 import Utils hiding (dbg, ptrace)
 import qualified Utils
 import Preprocessor
-
+import System.IO hiding (stdin)
 
 
 parseFromFileWithPreprocessor :: (Parser [ShellTest]) -> PreProcessor -> FilePath -> IO (Either ParseError [ShellTest])
-parseFromFileWithPreprocessor p preproc fname = do
-       input <- readFile fname
-       let processed = preprocess preproc input
-       case processed of Right text -> return (runParser p () fname text)
-                         (Left err)  -> return (Left err) -- To make haskell happy.
+parseFromFileWithPreprocessor p preproc fname =
+    do
+        h <- openFile fname ReadMode
+        hSetNewlineMode h universalNewlineMode
+        input <- hGetContents h
+        let processed = preprocess preproc input
+        case processed of    Right text -> return (runParser p () fname text)
+                             (Left err) -> return (Left err) -- To make haskell happy.
```

Comment:
I'm not sure what this means to me as a user - how would you explain more clearly in the manual or release notes ?

Reply:
Do You mean the line 25 or the way the line breaks are handled?

Reply:
I mean in general, what is the new functionality this provides.

Reply:
For user it gives macros (-D) and this is documented in CHANGES. For developer it gives opportunity to develop some other preprocessors.